README.rst

```diff
@@ -351,6 +352,48 @@ mode (see **Using transparent mode** above) or for a specific request (see
 **Sending requests with automatically-mapped parameters** above), Zyte API
 parameters are chosen as follows by default:
 
+-   ``Request.url`` becomes ``url``, same as in requests with manually-defined
+    parameters.
+
+-   If ``Request.method`` is something other than ``"GET"``, it becomes
+    ``httpRequestMethod``.
+
+-   ``Request.headers`` become ``customHttpRequestHeaders``.
+
+-   ``Request.body`` becomes ``httpRequestBody``.
+
+-   If the COOKIES_ENABLED_ Scrapy setting is ``True`` (default), and unless
+    request metadata sets dont_merge_cookies_ to ``True`` as well:
+
+    .. _COOKIES_ENABLED: https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#std-setting-COOKIES_ENABLED
+    .. _dont_merge_cookies: https://docs.scrapy.org/en/latest/topics/request-response.html#std-reqmeta-dont_merge_cookies
+
+    -   ``experimental.responseCookies`` is set to ``True`` .
+
+    -   Cookies from the request `cookie jar`_ become
+        ``experimental.requestCookies``.
+
+        .. _cookie jar: https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#std-reqmeta-cookiejar
+
+        All cookies from the cookie jar are set, regardless of their cookie
+        domain. This is because Zyte API requests may involve requests to
+        different domains (e.g. when following cross-domain redirects, or
+        during browser rendering).
+
+        If the cookies to be set exceed the limit defined in the
+        ``ZYTE_API_MAX_COOKIES`` setting (20 by default), a warning is logged,
+        and no cookies are set at all for the target request. To silence this
+        warning, set ``experimental.requestCookies`` manually, e.g. to an empty
+        dict. Alternatively, if Zyte API starts supporting more than 20 request
+        cookies, update the ``ZYTE_API_MAX_COOKIES`` setting accordingly.
```

Comment:
I am really unsure about what to do here. At one point I decided to remove this check altogether and do like we do with mixing httpResponseBody and browserHtml, i.e. let users send the wrong thing and let Zyte API report the issue.

However, mixing those 2 outputs never happens with plain automap, whereas the cookie limit can easily happen.

But even if we agree on keeping some form of handling on the client side, I am not sure what the best handling is. The proposal here logs a warning and sets no cookies at all. But that can often mean breaking the request. Setting 20 cookies could be just as bad. I wonder if it would not be better to directly raise an exception and not send the request at all.

Reply:
If cookies are dropped, there is a chance request still may work. If we raise an exception, it seems that there is no workaround for the user. So, it seems it'd be better to have a warning, not an exception. What do you think?

The limit is going to be increased to 100 soon; it's not a complete solution, but, hopefully, it'd help a bit.

Reply:
Hm, I was thinking that the warning is going to be issued, but the cookies which are under the limit are going to be sent. What's the advantage of dropping them all?

Reply:
No user-facing advantage.

If sending some cookies, though, I think we should discuss:
- Prioritization: should we prioritize some cookies over other, e.g. based on the target domain or how long the cookies have been in the cookie jar (e.g. the more recent, the higher the priority)?
- Communication: when dropping cookies, should we indicate, e.g. through a log message, which cookies are sent or which cookies are dropped? Or at least how many?